### PR TITLE
detect-pcre/v4: add tests

### PR DESCRIPTION
**Pull request by TheKharleeci**
Test conversion:
detect-pcre-01: DetectPcreModifPTest04
detect-pcre-02: DetectPcreModifPTest05
detect-pcre-03: DetectPcreTestSig01-03
detect-pcre-04: DetectPcreTestSig09-14
detect-pcre-05: DetectPcreTestSig15-16, DetectPcreFlowvarCapture01- 03

Previous PR: #776 

Corresponding Suricata PR: OISF/suricata/pull/6942

### CODE REVIEW

File: tests/detect-pcre/detect-pcre-02/writepcap.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+pkts = []
+
+load_layer("http")
+pkts += Ether(dst='ff:ff:ff:ff:ff:ff', src='00:01:02:03:04:05')/ \
+    Dot1Q(vlan=6)/ \
+    IPv6(dst='1.2.3.4', src='5.6.7.8')/TCP(sport=6666, dport=63, flags='P''A')/HTTP()/HTTPRequest(Method='GET', Path=' / ', Http_Version='HTTP/1.1', Host='www.emergingthreats.net', User_Agent='Mozilla/1.0', Content_Type='text/html; charset=utf-8\r\n\r\n15\r\n<!DOC')
+pkts += Ether(dst='ff:ff:ff:ff:ff:ff', src='00:01:02:03:04:05')/ \
+    Dot1Q(vlan=6)/ \
+    IPv6(dst='1.2.3.4', src='5.6.7.8')/TCP(sport=6666, dport=63, flags='P''A')/HTTP()/HTTPRequest(Content_Type='<TYPE html PUBLIC')
```
Comment (inashivb) [inline tests/detect-pcre/detect-pcre-02/writepcap.py:12]: Why is there a `<` here? Also, I do not think that `Content_Type` is the right key here..
It seems more like content that follows the type.. the beginning of an html page. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type

Comment (TheKharleeci) [reply tests/detect-pcre/detect-pcre-02/writepcap.py:12]: I was trying to create fragmented chunks. I think I'm doing it the wrong way as it doesn't generate the expected alert

**Comment by catenacyber** [comment]
Is this PR still live ?

**Comment by jufajardini** [comment]
Created an individual ticket to keep track of this work: https://redmine.openinfosecfoundation.org/issues/6147

I'll close this for now, Modupe has told me that she'd prefer to focus on the detect-lua PR, since she has less free time, now.

**Pull request closed without merging**